go/vt/sqlparser/parsed_query.go

```diff
@@ -155,9 +156,8 @@ func (pq *ParsedQuery) AppendFromRow(buf *bytes2.Buffer, fields []*querypb.Field
 }
 
 // MarshalJSON is a custom JSON marshaler for ParsedQuery.
-// Note that any queries longer that 512 bytes will be truncated.
 func (pq *ParsedQuery) MarshalJSON() ([]byte, error) {
-	return json.Marshal(TruncateForUI(pq.Query))
+	return json.Marshal(pq.Query)
```

Comment:
I have removed the truncation here. The JSON serialization for this was only used in tests and there we can also work fine without the truncation. Since it's not used outside of tests it looks like, it's safe here to remove. Keeping this was really annoying with how global state is affecting this, so removing it is far more trivial.